### PR TITLE
[SELC-4423] Fix: Added method to get institution id by giving priority to PT in case of double taxCode

### PR DESCRIPTION
**Pull request by empassaro**
#### List of Changes

Updated the createDelegation api by modifying the method to get the institution id, which in case there are multiple institutions with the same taxCode tries to get the institution with institutionType PT. If no PT is present, it fetches the first institution

#### Motivation and Context

This fix allows Institutions that create delegations for the prod-pagopa product to associate the delegation with the most appropriate institution type, which in this case is always the PT

#### How Has This Been Tested?

local environment

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java [hunk 1]
```diff
@@ -44,10 +45,36 @@ class DelegationServiceImplTest {
...
+        assertNotNull(response.getId());
+        assertEquals(delegation.getId(), response.getId());
+    }
+
+    /**
+     * Method under test: {@link DelegationServiceImpl#createDelegation(Delegation)}
+     */
+    @Test
+    void testCreateDelegationForPTWithProductPagopa() {
+        Delegation delegation = new Delegation();
```
Comment (pierpaolodidato89) [inline core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java:66]: We could avoid to duplicate creation of delegation object in each method creating a static dummy object.

Comment (empassaro) [reply core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java]: done

File: core/src/main/java/it/pagopa/selfcare/mscore/core/DelegationServiceImpl.java [hunk 2]
```diff
@@ -41,18 +40,15 @@ public DelegationServiceImpl(DelegationConnector delegationConnector,
 
     @Override
     public Delegation createDelegation(Delegation delegation) {
+        if(PROD_PAGOPA.equals(delegation.getProductId())) {
```
Comment (manuraf) [inline core/src/main/java/it/pagopa/selfcare/mscore/core/DelegationServiceImpl.java:43]: you could add some comments to explain why this behaviour

Comment (empassaro) [reply core/src/main/java/it/pagopa/selfcare/mscore/core/DelegationServiceImpl.java:47]: Done

**Pull request merged**